ietf/meeting/tests_js.py

```diff
@@ -283,6 +284,254 @@ def test_edit_meeting_schedule(self):
         self.assertTrue(self.driver.find_elements_by_css_selector('#timeslot{} #session{}'.format(slot1b.pk, s1.pk)),
                         'Session s1 should have moved to second timeslot on first meeting day')
 
+    def test_past_flags(self):
+        """Test that timeslots and sessions in the past are marked accordingly
+
+        Would also like to test that past-hint flags are applied when a session is dragged, but that
+        requires simulating HTML5 drag-and-drop. Have not yet found a good way to do this.
+        """
+        wait = WebDriverWait(self.driver, 2)
+        meeting = MeetingFactory(type_id='ietf')
+        room = RoomFactory(meeting=meeting)
+
+        # get current time in meeting time zone
+        right_now = now().astimezone(
+            pytz.timezone(meeting.time_zone)
+        )
+        if not settings.USE_TZ:
+            right_now = right_now.replace(tzinfo=None)
+
+        past_timeslots = [
+            TimeSlotFactory(meeting=meeting, time=right_now - datetime.timedelta(hours=n),
+                            duration=datetime.timedelta(hours=1), location=room)
+            for n in range(1,4)
+        ]
+        future_timeslots = [
+            TimeSlotFactory(meeting=meeting, time=right_now + datetime.timedelta(hours=n),
+                            duration=datetime.timedelta(hours=1), location=room)
+            for n in range(1,4)
+        ]
```

Comment:
I would be interested to check a timeslot currently happening as well.

Reply:
Ok - I'm putting one just barely in the past and another a few minutes in the future to avoid the possibility of race conditions.